Output filter doesn't support fields with underscore or hyphen

Quite a lot of fields contains underscore/hyphens, e.g. `saddr-raw`, `icmp_responder`, and the field I especially wanted to use: `udp_pkt_size`.

But the token `T_FIELD`  in the filter grammar only matches alphanumeric characters, thus these fields cannot be used in output filters.

Just an example of wanted behavior:

Before patch:
```bash
# zmap --quiet -M udp -p 53 --output-filter="udp_pkt_size > 10" -f saddr,data -N 1
Parse error: syntax error
Sep 15 21:35:29.797 [ERROR] zmap: Unable to parse filter string: 'udp_pkt_size > 10'
Sep 15 21:35:29.797 [FATAL] zmap: Unable to parse filter expression
```

After:
```bash
# /src/zmap --quiet -M udp -p 53 --output-filter="udp_pkt_size > 10" -f saddr,data -N 1
Sep 15 21:35:00.297 [INFO] zmap: output module: csv
Sep 15 21:35:00.297 [INFO] csv: no output file selected, will use stdout
saddr,data
178.33.XX.XXX,4745d0010000000000000000
Sep 15 21:35:01.418 [INFO] zmap: completed
```

--- a/src/lexer.l
+++ b/src/lexer.l
@@ -31,6 +31,6 @@
 "<"                  return '<';
 "("                  return '(';
 ")"                  return ')';
-[a-zA-Z][a-zA-Z0-9]+ yylval.string_literal = strdup(yytext); return T_FIELD;
+[a-zA-Z][-_a-zA-Z0-9]+ yylval.string_literal = strdup(yytext); return T_FIELD;
 
 %%

